boa/src/builtins/value/mod.rs

```diff
@@ -34,11 +35,77 @@ pub use operations::*;
 pub type ResultValue = Result<Value, Value>;
 
 /// A Garbage-collected Javascript value as represented in the interpreter.
-pub type Value = Gc<ValueData>;
+#[derive(Debug, Clone, Trace, Finalize, Default)]
+pub struct Value(pub(crate) Gc<ValueData>);
 
-#[inline]
-pub fn undefined() -> Value {
-    Gc::new(ValueData::Undefined)
+impl Value {
+    #[inline]
+    pub fn undefined() -> Self {
+        Self(Gc::new(ValueData::Undefined))
+    }
+
+    #[inline]
+    pub fn null() -> Self {
+        Self(Gc::new(ValueData::Null))
+    }
+
+    #[inline]
+    pub fn string(value: String) -> Self {
+        Self(Gc::new(ValueData::String(value)))
+    }
```

Comment:
I think we should use `Into<String>` here, so that we can avoid all those `to_string()` when calling it, and we can just write `Value::string("hello")`. This will be a no-op for a `String`.

```suggestion
    pub fn string<S>(value: S) -> Self where S: Into<String> {
        Self(Gc::new(ValueData::String(value.into())))
    }
```